File: Ejercicios Propuestos 1/Ejercicio 16/Ejercicio16.java
import java.util.Scanner;
public class Ejercicio16 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in); 
        System.out.print("Ingrese un número natural positivo: ");
        int numero = scanner.nextInt();   
        int suma = 0;
        for (int i = 1; i < numero; i++) {
            if (numero % i == 0) {
                suma += i;
            }
        }
        if (suma == numero) {
            System.out.println(numero + " es un número perfecto");
        } else {
            System.out.println(numero + " NO es un número perfecto");
        }
    }
}

File: Ejercicios Propuestos 1/Ejercicio 16/Ejercicio 16.jl
print("Introduce un número entero positivo: ")
n = parse(Int, readline())

suma_divisores = 0

for i in 1:(n - 1)
    if n % i == 0
        global suma_divisores += i
    end
end

if suma_divisores == n
    println("El número ", n, " es perfecto.")
else
    println("El número ", n, " no es perfecto.")
end
